cmd/core-data/main.go

```diff
@@ -49,7 +52,17 @@ func main() {
 	flag.Usage = usage.HelpCallback
 	flag.Parse()
 
-	httpServer := httpserver.NewBootstrap(data.LoadRestRoutes())
+	configuration := &data.ConfigurationStruct{}
+	dic := di.NewContainer(di.ServiceConstructorMap{
+		container.ConfigurationName: func(get di.Get) interface{} {
+			return configuration
+		},
+		bootstrapContainer.ConfigurationInterfaceName: func(get di.Get) interface{} {
+			return get(container.ConfigurationName)
+		},
+	})
```

Comment:
You need a new empty container here.  

bootstrap.Run() will populate ConfigurationInterfaceName.  

ConfigurationName isn't used elsewhere within your code and should be dropped.  You'll likely add these as part of #2031.